src/checker.rs

```diff
@@ -3,6 +3,28 @@ use regex::bytes::Regex;
 
 use crate::*;
 
+macro_rules! mk_else_if {
+    ($if_type: ident, $block_type: ident) => {
+        #[inline(always)]
+        fn is_else_if(node: &Node) -> bool {
+            if node.object().kind_id() != <Self as TSLanguage>::BaseLang::$if_type {
+                return false;
+            }
+            if let Some(parent) = node.object().parent() {
+                if parent.kind_id() == <Self as TSLanguage>::BaseLang::ElseClause {
+                    return true;
+                }
+                if let Some(grandparent) = parent.parent() {
```

Comment:
I really wonder if it's good way to do...
To be clear I don't say that this code is bad but I've just a bad feeling (and maybe I'm totally wrong).
I need think about that.

Reply:
Yes, the implementation should better consider how an AST is visited, but a different one would require to fix #289 first. Since we have added tests which seem to work correctly, I propose to postpone a more correct implementation at a later time.